From cf54e8c4476336e1ea293387fe837f3bba5cb743 Mon Sep 17 00:00:00 2001
From: Felix <8057646+GoliathLabs@users.noreply.github.com>
Date: Sat, 17 Feb 2024 15:06:26 +0100
Subject: [PATCH] Update ramips-mt7621

---
 targets/ramips-mt7621 | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/targets/ramips-mt7621 b/targets/ramips-mt7621
index 01b9b94cdf..442484853a 100644
--- a/targets/ramips-mt7621
+++ b/targets/ramips-mt7621
@@ -39,6 +39,8 @@ device('d-link-covr-x1860-a1', 'dlink_covr-x1860-a1', {
 	},
 })
 
+device('d-link-dap-1620-b1', 'dlink_dap-1620-b1')
+
 device('d-link-dap-x1860-a1', 'dlink_dap-x1860-a1')
 
 device('d-link-dir-860l-b1', 'dlink_dir-860l-b1')
