From 59b0d58c9c7ed102b224c795fadeef1f48d81daf Mon Sep 17 00:00:00 2001
From: Felix <8057646+GoliathLabs@users.noreply.github.com>
Date: Mon, 25 Dec 2023 00:36:47 +0100
Subject: [PATCH] ramips-mt7621: add dlink-dap-1620-b1

---
 targets/ramips-mt7621 | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/targets/ramips-mt7621 b/targets/ramips-mt7621
index c5708838c1..9ff718d08f 100644
--- a/targets/ramips-mt7621
+++ b/targets/ramips-mt7621
@@ -33,6 +33,8 @@ device('cudy-x6-v2', 'cudy_x6-v2', {

 -- D-Link

+device('d-link-dap-1620-b1', 'dlink_dap-1620-b1')
+
 device('d-link-dap-x1860-a1', 'dlink_dap-x1860-a1')

 device('d-link-dir-860l-b1', 'dlink_dir-860l-b1')
