From 52a942ac50b33d087f1104197a55baa473906d13 Mon Sep 17 00:00:00 2001
From: Grische <2787581+grische@users.noreply.github.com>
Date: Sun, 5 Jan 2025 17:00:59 +0100
Subject: [PATCH] targets: remove more devices with small flash

---
 targets/ramips-mt7620 | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/targets/ramips-mt7620 b/targets/ramips-mt7620
index c251c51a..65bdc945 100644
--- a/targets/ramips-mt7620
+++ b/targets/ramips-mt7620
@@ -37,20 +37,6 @@ if (env.GLUON_REGION or '') ~= '' then
 	tplink_region_suffix = '-' .. env.GLUON_REGION
 end
 
-device('tp-link-archer-c2-v1', 'tplink_archer-c2-v1', {
-	factory = false,
-})
-
-device('tp-link-archer-c20-v1', 'tplink_archer-c20-v1', {
-	factory = false,
-})
-
-device('tp-link-archer-c20i', 'tplink_archer-c20i')
-
-device('tp-link-archer-c50-v1', 'tplink_archer-c50-v1', {
-	factory = '-squashfs-factory' .. tplink_region_suffix,
-})
-
 
 -- Xiaomi
 
-- 
2.34.1

