diff --git a/targets/bcm27xx-bcm2711 b/targets/bcm27xx-bcm2711
index 84fb1087..e2ec0841 100644
--- a/targets/bcm27xx-bcm2711
+++ b/targets/bcm27xx-bcm2711
@@ -2,6 +2,8 @@ include 'bcm27xx.inc'
 
 device('raspberrypi-4-model-b', 'rpi-4', {
 	manifest_aliases = {
+		-- from Gluon v2023.1 and older for ffmuc only
+		'raspberry-pi-4-model-b',
 		-- from Gluon 2023.1 and older
 		'raspberry-pi-4-model-b-rev-1.1',
 		'raspberry-pi-4-model-b-rev-1.2', -- untested
