diff --git a/targets/sunxi-cortexa7 b/targets/sunxi-cortexa7
index f6f56248..1ac66da5 100644
--- a/targets/sunxi-cortexa7
+++ b/targets/sunxi-cortexa7
@@ -15,3 +15,7 @@ device('lemaker-banana-pro', 'lemaker_bananapro', {
 device('lamobo-r1', 'lamobo_lamobo-r1', {
        broken = true, -- AP+11s not working
 })
+
+device('friendlyelec-nanopi-r1', 'friendlyarm_nanopi-r1', {
+       factory = false,
+})
