--- a/package/gluon-core/luasrc/lib/gluon/upgrade/010-primary-mac
+++ b/package/gluon-core/luasrc/lib/gluon/upgrade/010-primary-mac
@@ -119,6 +119,9 @@ local primary_addrs = {
                        'openmesh,a42',
                        'openmesh,a62',
                }},
+               {'kirkwood', 'generic', {
+                       'linksys,e4200-v2',
+               }},
                {'mpc85xx', 'p1020', {
                        'aerohive,hiveap-330',
                        'ocedo,panda',
--- /dev/null
+++ b/targets/kirkwood-generic
@@ -0,0 +1,5 @@
+-- Linksys
+
+device('linksys-e4200-v2', 'linksys_e4200-v2', {
+	broken = true,
+})
--- a/targets/targets.mk
+++ b/targets/targets.mk
@@ -29,5 +29,6 @@ $(eval $(call GluonTarget,x86,64))
 ifeq ($(BROKEN),1)
 $(eval $(call GluonTarget,bcm27xx,bcm2710)) # BROKEN: Untested
 $(eval $(call GluonTarget,bcm27xx,bcm2711)) # BROKEN: No 11s support, no reset button, sys LED issues
+$(eval $(call GluonTarget,kirkwood,generic))
 $(eval $(call GluonTarget,mvebu,cortexa9)) # BROKEN: No 11s support
 endif
