diff --git a/targets/bcm27xx-bcm2711 b/targets/bcm27xx-bcm2711
new file mode 100644
index 00000000..ba631911
--- /dev/null
+++ b/targets/bcm27xx-bcm2711
@@ -0,0 +1,9 @@
+include 'bcm27xx.inc'
+
+device('raspberry-pi-4-model-b', 'rpi-4', {
+	manifest_aliases = {
+		'raspberry-pi-4-model-b-rev-1.1',
+		'raspberry-pi-4-model-b-rev-1.2',
+		'raspberry-pi-4-model-b-rev-1.4',
+	},
+})
diff --git a/targets/targets.mk b/targets/targets.mk
index 92b58ab8..05a74935 100644
--- a/targets/targets.mk
+++ b/targets/targets.mk
@@ -22,5 +22,6 @@ $(eval $(call GluonTarget,x86,64))
 
 ifneq ($(BROKEN),)
 $(eval $(call GluonTarget,bcm27xx,bcm2710)) # BROKEN: Untested
+$(eval $(call GluonTarget,bcm27xx,bcm2711)) # BROKEN: Untested
 $(eval $(call GluonTarget,mvebu,cortexa9)) # BROKEN: No 11s support
 endif
