diff --git a/targets/ipq40xx-generic b/targets/ipq40xx-generic
index e3fc746b..d5194a37 100644
--- a/targets/ipq40xx-generic
+++ b/targets/ipq40xx-generic
@@ -50,6 +50,7 @@ device('avm-fritz-box-4040', 'avm_fritzbox-4040', {
 
 device('avm-fritz-box-7530', 'avm_fritzbox-7530', {
 	factory = false,
+	aliases = {'avm-fritz-box-7520'},
 })
 
 device('avm-fritz-repeater-1200', 'avm_fritzrepeater-1200', {
