From e6c662741d0ec06c15b0b4517a7ddd85e051bc6b Mon Sep 17 00:00:00 2001
From: Grische <2787581+grische@users.noreply.github.com>
Date: Sun, 5 Jan 2025 14:38:08 +0100
Subject: [PATCH] temporary remove devices with too little flash

---
 targets/ath79-generic | 15 ---------------
 targets/ramips-mt7620 | 16 ----------------
 2 files changed, 31 deletions(-)

diff --git a/targets/ath79-generic b/targets/ath79-generic
index 1d7f7348..a23e0564 100644
--- a/targets/ath79-generic
+++ b/targets/ath79-generic
@@ -432,10 +432,6 @@ device('tp-link-archer-c58-v1', 'tplink_archer-c58-v1', {
 	class = 'tiny', -- 64M ath9k + ath10k
 })
 
-device('tp-link-archer-c6-v2-eu-ru-jp', 'tplink_archer-c6-v2', {
-	packages = ATH10K_PACKAGES_QCA9888,
-})
-
 device('tp-link-archer-c7-v2', 'tplink_archer-c7-v2', {
 	packages = ATH10K_PACKAGES_QCA9880,
 	factory = '-squashfs-factory' .. tplink_region_suffix,
@@ -453,11 +449,6 @@ device('tp-link-archer-c59-v1', 'tplink_archer-c59-v1', {
 	packages = ATH10K_PACKAGES_QCA9888,
 })
 
-device('tp-link-archer-c60-v1', 'tplink_archer-c60-v1', {
-	packages = ATH10K_PACKAGES_SMALLBUFFERS_QCA9880,
-	factory = false,
-})
-
 device('tp-link-archer-d50-v1', 'tplink_archer-d50-v1', {
 	packages = ATH10K_PACKAGES_SMALLBUFFERS_QCA9880,
 	factory = false,
@@ -498,12 +489,6 @@ device('tp-link-tl-wr810n-v1', 'tplink_tl-wr810n-v1')
 
 device('tp-link-tl-wr842n-v3', 'tplink_tl-wr842n-v3')
 
-device('tp-link-tl-wr902ac-v1', 'tplink_tl-wr902ac-v1', {
-	packages = ATH10K_PACKAGES_SMALLBUFFERS_QCA9887,
-	broken = true, -- OOM with 5GHz enabled in most environments
-	class = 'tiny', -- 64M ath9k + ath10k
-})
-
 device('tp-link-tl-wr1043nd-v2', 'tplink_tl-wr1043nd-v2')
 
 device('tp-link-tl-wr1043nd-v3', 'tplink_tl-wr1043nd-v3')
diff --git a/targets/ramips-mt7620 b/targets/ramips-mt7620
index 36940042..c251c51a 100644
--- a/targets/ramips-mt7620
+++ b/targets/ramips-mt7620
@@ -30,22 +30,6 @@ device('netgear-ex3700', 'netgear_ex3700', {
 	factory_ext = '.chk',
 })
 
-device('netgear-ex6130', 'netgear_ex6130', {
-	factory_ext = '.chk',
-})
-
-
--- Nexx
-
-device('nexx-wt3020-8m', 'nexx_wt3020-8m', {
-	aliases = {
-		'nexx-wt3020ad',
-		'nexx-wt3020f',
-		'nexx-wt3020h',
-	},
-})
-
-
 -- TP-Link
 
 local tplink_region_suffix = ''
-- 
2.34.1

